elasticsearch-transport/lib/elasticsearch/transport/transport/http/manticore.rb

```diff
@@ -43,97 +46,156 @@ module HTTP
+                    logger.warn "[#{e.class}] Reloading connections (attempt #{tries} of #{connections.all.size})" if logger
+                    reload_connections!
+                  end
 
-                Connections::Connection.new \
-                  :host => host,
-                  :connection => @manticore
-              },
-              :selector_class => options[:selector_class],
-              :selector => options[:selector]
+                  retry
+                else
+                  logger.fatal "[#{e.class}] Cannot connect to #{connection.host.inspect} after #{tries} tries" if logger
+                  raise e
+                end
+              end
+            rescue Exception => e
+              logger.fatal "[#{e.class}] #{e.message} ()" if logger
+              raise e
+            end
           end
 
-          # Closes all connections by marking them as dead
-          # and closing the underlying HttpClient instances
-          #
-          # @return [Connections::Collection]
-          #
           def __close_connections
-            @connections.each {|c| c.dead! }
-            @connections.all.each {|c| c.connection.close }
+            if @sniffer
+              logger.info("Closing sniffer...") if logger
+              @sniffer.close
+            end
+            logger.info("Sniffer closed.") if logger
+            logger.info("Closing pool") if logger
+            @pool.close # closes adapter as well
           end
 
-          # Returns an array of implementation specific connection errors.
-          #
-          # @return [Array]
-          #
-          def host_unreachable_exceptions
-            [
-              ::Manticore::Timeout,
-              ::Manticore::SocketException,
-              ::Manticore::ClientProtocolException,
-              ::Manticore::ResolutionFailure
-            ]
+          def enriching_response(method, path, params, body)
+            start = Time.now if logger || tracer
+
+            url, response = yield
+
+            duration = Time.now-start if logger || tracer
+
+            if response.status.to_i >= 300
+              __log    method, path, params, body, url, response, nil, 'N/A', duration if logger
+              __trace  method, path, params, body, url, response, nil, 'N/A', duration if tracer
+              __log_failed response if logger
+              __raise_transport_error response
+            end
+
+            json = deserialize_response(response)
+
+            took     = (json['took'] ? sprintf('%.3fs', json['took']/1000.0) : 'n/a') rescue 'n/a' if logger || tracer
+
+            __log   method, path, params, body, url, response, json, took, duration if logger
+            __trace method, path, params, body, url, response, json, took, duration if tracer
+
+
+            ::Elasticsearch::Transport::Transport::Response.new response.status, json || response.body, response.headers
```

Comment:
not sure I understand the `json || response.body` part. in what situation don't we want to use the output of deseralize_response ?